### PR TITLE
Migrate device-state tests to integration

### PR DESCRIPTION
**Pull request by pipex**
Change-type: patch

### CODE REVIEW

**Review by cywang117 — COMMENTED** [review]
Overall LGTM, great work as usual @pipex !

File: src/device-state/preload.ts [hunk 1]
```diff
@@ -130,7 +130,7 @@ export async function loadTargetFromFile(appsPath: string): Promise<boolean> {
 		const localState = {
 			[uuid]: {
 				name: '',
-				config: { ...formattedConf, ...deviceConf },
+				config: { ...deviceConf, ...formattedConf },
```
[the remark this answers is not in the document]
Comment (pipex) [reply src/device-state/preload.ts:133]: I have not found any, my suspicion is that preloading config vars is not that common, which is why we have not seen anyone affected by this.

File: test/lib/dbus/systemd.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+		}
+
+		const { path } = units[unitName];
+		callback(null, path);
+	},
+);
+
+// Simulate OS units
+createUnit('openvpn.service', '/org/freedesktop/systemd1/unit/openvpn');
+createUnit('avahi.socket', '/org/freedesktop/systemd1/unit/avahi');
```
[the remark this answers is not in the document]
Comment (pipex) [reply test/lib/dbus/systemd.ts:171]: The way I see it is that we really want to test against a specific type of interface, so this way isolates the interface without bringing any additional behavior from balenaOS. I'd be willing to switch to balenaOS in the future if it makes sense.

[the remark this answers is not in the document]
Comment (cywang117) [reply test/lib/dbus/systemd.ts:171]: Makes sense for unit/integration tests. Using the actual OS would be better for if we ever have E2E tests (aside from the existing OS tests).

**Comment by pipex** [comment]
@balena-ci I self certify!

**Pull request merged**